datafusion/datasource-csv/src/source.rs

```diff
@@ -617,8 +617,13 @@ impl FileSource for CsvSource {
     fn file_type(&self) -> &str {
         "csv"
     }
-    fn fmt_extra(&self, _t: DisplayFormatType, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, ", has_header={}", self.has_header)
+    fn fmt_extra(&self, t: DisplayFormatType, f: &mut fmt::Formatter) -> fmt::Result {
+        match t {
+            DisplayFormatType::Default | DisplayFormatType::Verbose => {
+                write!(f, ", has_header={}", self.has_header)
+            }
+            DisplayFormatType::TreeRender => Ok(()),
```

Comment:
Per the description of TreeRender:
https://github.com/apache/datafusion/blob/3dc212c9078c92f57ab7f58e75e1258130c772d0/datafusion/physical-plan/src/display.rs#L48-L74

TreeRender mode should have only the most relevant details for understanding the high level plan